agent/engine/engine_windows_integ_test.go

```diff
@@ -611,3 +622,152 @@ func TestTaskLevelVolume(t *testing.T) {
+	require.NoError(t, err)
+
+	client, _ := sdkClient.NewClientWithOpts(sdkClient.WithHost(endpoint), sdkClient.WithVersion(sdkclientfactory.GetDefaultVersion().String()))
+	containerMap, _ := taskEngine.(*DockerTaskEngine).state.ContainerMapByArn(testTask.Arn)
+	cid := containerMap[testTask.Containers[0].Name].DockerID
+
+	err = verifyContainerCredentialSpec(client, cid)
+	assert.NoError(t, err)
+
+	// Kill the existing container now
+	err = client.ContainerKill(context.TODO(), cid, "SIGKILL")
+	assert.NoError(t, err, "Could not kill container")
+
+	verifyTaskIsStopped(stateChangeEvents, testTask)
+
+	// Cleanup the test file
+	err = os.Remove(testCredSpecFilePath)
+	assert.NoError(t, err)
+}
+
+func defaultTestConfigIntegTestGMSA() *config.Config {
+	cfg, _ := config.NewConfig(ec2.NewBlackholeEC2MetadataClient())
+	cfg.TaskCPUMemLimit = config.ExplicitlyDisabled
+	cfg.ImagePullBehavior = config.ImagePullPreferCachedBehavior
+
+	cfg.GMSACapable = true
+	cfg.AWSRegion = "us-west-2"
+
+	return cfg
+}
+
+func setupGMSA(cfg *config.Config, state dockerstate.TaskEngineState, t *testing.T) (TaskEngine, func(), credentials.Manager) {
+	ctx, cancel := context.WithCancel(context.TODO())
+	defer cancel()
+
+	if os.Getenv("ECS_SKIP_ENGINE_INTEG_TEST") != "" {
+		t.Skip("ECS_SKIP_ENGINE_INTEG_TEST")
+	}
+	if !isDockerRunning() {
+		t.Skip("Docker not running")
+	}
+
+	sdkClientFactory := sdkclientfactory.NewFactory(ctx, dockerEndpoint)
+	dockerClient, err := dockerapi.NewDockerGoClient(sdkClientFactory, cfg, context.Background())
+	if err != nil {
+		t.Fatalf("Error creating Docker client: %v", err)
+	}
+	credentialsManager := credentials.NewManager()
+	if state == nil {
+		state = dockerstate.NewTaskEngineState()
+	}
+	imageManager := NewImageManager(cfg, dockerClient, state)
+	imageManager.SetSaver(statemanager.NewNoopStateManager())
+	metadataManager := containermetadata.NewManager(dockerClient, cfg)
+
+	resourceFields := &taskresource.ResourceFields{
+		ResourceFieldsCommon: &taskresource.ResourceFieldsCommon{
+			SSMClientCreator: ssmfactory.NewSSMClientCreator(),
+		},
+		DockerClient:    dockerClient,
+		S3ClientCreator: s3factory.NewS3ClientCreator(),
+	}
+
+	taskEngine := NewDockerTaskEngine(cfg, dockerClient, credentialsManager,
+		eventstream.NewEventStream("ENGINEINTEGTEST", context.Background()), imageManager, state, metadataManager, resourceFields)
+	taskEngine.MustInit(context.TODO())
+	return taskEngine, func() {
+		taskEngine.Shutdown()
+	}, credentialsManager
+}
+
+func setupWithGMSA(t *testing.T) (TaskEngine, func(), credentials.Manager) {
+	return setupGMSA(defaultTestConfigIntegTestGMSA(), nil, t)
+}
+
+func verifyContainerCredentialSpec(client *sdkClient.Client, id string) error {
+	dockerContainer, err := client.ContainerInspect(context.TODO(), id)
+	if err != nil {
+		return err
+	}
+
+	for _, opt := range dockerContainer.HostConfig.SecurityOpt {
+		if strings.HasPrefix(opt, "credentialspec=") {
```

Comment:
why we don't directly check the credentialspec value here?

Reply:
Updated helper.